lib/cfme/cloud_services/inventory_sync.rb

```diff
@@ -29,6 +29,27 @@ def self.sync_queue(userid, targets)
         MiqTask.generic_action_with_callback(task_opts, queue_opts)
       end
 
+      def self.collect(opts)
+        path = DataPackager.package(DataCollector.collect(opts[:manifest], targets_from_queue(opts[:targets])))
+        update_task(opts[:task_id], path.to_s) if opts[:task_id]
+        path
+      end
+
+      def self.collect_queue(userid, manifest, targets)
+        task_opts = {
+          :action => "Collect and package inventory",
+          :userid => userid
+        }
+
+        queue_opts = {
+          :class_name  => name,
+          :method_name => "collect",
+          :args        => [{:manifest => manifest, :targets => targets_for_queue(targets)}]
```

Comment:
Any reason we need to use an options hash (or kwargs)?  Can it just be normal parameters?

Reply:
If the args aren't a hash the task_id won't be added in automatically